### ISSUE TITLE
native-image agent records wrong signature in Jettys TypeUtil

### ISSUE THREAD

**Issue by mhalbritter** [oracle/graal#5204]
Jetty has a class named `TypeUtil`, [see here](https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/jetty-util/src/main/java/org/eclipse/jetty/util/TypeUtil.java#L195). In this class, they use `MethodHandles.Lookup` to get `MethodHandle`s to `getCodeSourceLocation`, etc.

When running a test which uses the `TypeUtil` with agent attached, I can see `getCodeSourceLocation` etc, but the signature is wrong:

```json
{
  "condition":{"typeReachable":"org.eclipse.jetty.util.TypeUtil"},
  "name":"org.eclipse.jetty.util.TypeUtil",
  "methods":[
    {"name":"getClassLoaderLocation","parameterTypes":[] }, 
    {"name":"getCodeSourceLocation","parameterTypes":[] }, 
    {"name":"getModuleLocation","parameterTypes":[] }, 
    {"name":"getSystemClassLoaderLocation","parameterTypes":[] }
  ]
}
```

The signatures are `URI` as return type and `Class<?>` as only argument, e.g. `public static URI getCodeSourceLocation(Class<?> clazz)`

Version info:

```
openjdk version "17.0.4" 2022-07-19
OpenJDK Runtime Environment GraalVM CE 22.2.0 (build 17.0.4+8-jvmci-22.2-b06)
OpenJDK 64-Bit Server VM GraalVM CE 22.2.0 (build 17.0.4+8-jvmci-22.2-b06, mixed mode, sharing)
```

**Comment by vjovanov** [comment]
@mhalbritter thanks greatly for reporting! This is indeed strange, do you have a reproducer at hand that we could use?

Seems to me that `com.oracle.svm.agent.BreakpointInterceptor#getParamTypes` somehow fails to fetch the correct types.

**Comment by vjovanov** [comment]
Need a full reproducer here. I get the following:
```json
{
  "name":"org.eclipse.jetty.util.TypeUtil",
  "methods":[
    {"name":"getClassLoaderLocation","parameterTypes":["java.lang.Class"] }, 
    {"name":"getCodeSourceLocation","parameterTypes":["java.lang.Class"] }, 
    {"name":"getModuleLocation","parameterTypes":["java.lang.Class"] }, 
    {"name":"getSystemClassLoaderLocation","parameterTypes":["java.lang.Class"] }
  ]
}
```

**Comment by mhalbritter** [comment]
Interesting. I tried to build a reproducer, and if I use jetty in a `main` method with attached agent, all calls are recorded.

I'm currently working on metadata for jetty, which runs jetty in a `nativeTest` via JUnit, and there the calls aren't recorded properly. Trying to build a minimal sample from that.

**Comment by vjovanov** [comment]
Might be that the merging of config introduces the error.

**Comment by mhalbritter** [comment]
Yes, i'm using agent conditional mode.

**Comment by gradinac** [comment]
Hey @mhalbritter! :) Hmm, that is odd, so if I get this right, it only fails when you run these tests in the metadata repository? Could it be that tests in the repository use an older version of `native-build-tools`? I remember we had some filtering problems where we unconditionally filtered all `java.*` entries: https://github.com/graalvm/native-build-tools/pull/289
If it's using the newest version, then we may be specifying this filter manually somewhere in the metadata repository

**Comment by mhalbritter** [comment]
Ah, thanks for the hint. The metadata repo indeed uses `0.9.13`, whereas my test uses `0.9.14`.

**Comment by gradinac** [comment]
I think this is the root cause of the issue - could you, if you have the time of course, as a test bump the version to 0.9.14 locally and see if it passes? I think it should as the fix for this issue got into 0.9.14 but not 0.9.13 - if it passes, we most likely need to bump the NBT version in the metadata repository

**Comment by mhalbritter** [comment]
Yes, can confirm that a NBT update fixes this issue.
